rtloader/common/stringutils.c

```diff
@@ -42,23 +42,34 @@ char *as_string(PyObject *object)
         // PyString_AsString might raise an error when python can't encode a
         // unicode string to byte
         PyErr_Clear();
-        return  NULL;
+        return NULL;
     }
     retval = strdupe(tmp);
 #else
-    if (!PyUnicode_Check(object)) {
-        return NULL;
-    }
-
-    PyObject *temp_bytes = PyUnicode_AsEncodedString(object, "UTF-8", "strict");
-    if (temp_bytes == NULL) {
-        // PyUnicode_AsEncodedString might raise an error if the codec raised an
-        // exception
-        PyErr_Clear();
-        return NULL;
+    PyObject *temp_bytes = NULL;
+
+    if (PyBytes_Check(object)) {
+        // We already have an encoded string, we suppose it has the correct encoding (UTF-8)
+        temp_bytes = object;
+        Py_INCREF(temp_bytes);
+    } else if (PyUnicode_Check(object)) {
+        // Encode the Unicode string that was given
+        temp_bytes = PyUnicode_AsEncodedString(object, "UTF-8", "strict");
+        if (temp_bytes == NULL) {
+            // PyUnicode_AsEncodedString might raise an error if the codec raised an
+            // exception
+            PyErr_Clear();
+            retval = NULL;
+            goto done;
+        }
+    } else {
+        retval = NULL;
+        goto done;
     }
 
     retval = strdupe(PyBytes_AS_STRING(temp_bytes));
+
+done:
```

Comment:
The two cases where we jump to `done`, could I think be replace by a `return NULL`. We could remove this label.